sdcm/utils/sstable_utils.py

```diff
@@ -0,0 +1,142 @@
+            self.log.debug('Failed to find sstables for %s: %s', self.ks_cf, error)
+
+        message = f'filtered by last {from_minutes_ago} minutes' if from_minutes_ago else '(not filtered by time)'
+        self.log.debug('Got %s sstables %s', len(selected_sstables), message)
+        return selected_sstables
+
+    def count_sstable_tombstones(self, sstable: str) -> int:
+        try:
+            self.db_node.remoter.run(f'sudo sstabledump  {sstable} 1>/tmp/sstabledump.json', verbose=False)
+            tombstones_deletion_info = self.db_node.remoter.run(
+                'sudo egrep \'"expired" : true|marked_deleted\' /tmp/sstabledump.json', verbose=False)
+            if not tombstones_deletion_info:
+                self.log.debug('Got no tombstones for sstable: %s', sstable)
+                return 0
+
+            num_tombstones = len(tombstones_deletion_info.stdout.splitlines())
+            self.log.debug('Got %s tombstones for sstable: %s', num_tombstones, sstable)
+            return num_tombstones
+        except Exception as error:  # pylint: disable=broad-except
+            self.log.error('count_sstable_tombstones failed with: %s', error)
+            return 0
+
+    def get_table_repair_date(self) -> str | None:
+        """
+        Search entries of requested table in system.repair_history
+        Return last entry found.
+        Example returned value: '2022-12-28 11:53:53'
+        """
+        with self.db_cluster.cql_connection_patient(node=self.db_node, connect_timeout=300,
+                                                    user=self.user, password=self.password) as session:
+            try:
+                query = f"SELECT repair_time from system.repair_history WHERE keyspace_name = '{self.keyspace}' " \
+                        f"AND table_name = '{self.table}' ALLOW FILTERING;"
+                results = session.execute(query)
+                output = results.all()
+                self.log.debug('SELECT repair_time results: %s', output)
+                if len(output) == 0:
+                    self.log.debug('No repair history found for %s.%s', self.keyspace, self.table)
+                    return None
+                return str(output[-1].repair_time)
+            except Exception as exc:  # pylint: disable=broad-except
+                self.log.warning('Failed to get repair date of %s.%s. Error: %s', self.keyspace, self.table, exc)
+                raise
+
+    def get_table_repair_date_and_delta_minutes(self) -> (datetime.datetime, int):
+
+        table_repair_date = self.get_table_repair_date()
+        table_repair_date = datetime.datetime.strptime(table_repair_date, '%Y-%m-%d %H:%M:%S')
+        now = datetime.datetime.now()
+        delta_repair_date_minutes = ((now - table_repair_date).seconds - self.propagation_delay_in_seconds) // 60
+        self.log.debug('Found table-repair-date: %s, Ended %s minutes ago (%s)', table_repair_date,
+                       delta_repair_date_minutes, now)
```

Comment:
i'm not sure you need to log `now`, as it will be already logged in the logger side, but if you do, be clear what that thing between parenthesis means

Reply:
fixed